### PR TITLE
Support disabling instance launcher "auto safe mode"

### LINKED ISSUE #4920: Update FlowFuse to present and persist disabling auto safe mode option

### PR DESCRIPTION
**Pull request by Steve-Mcl**
closes #4920 

## Description

Adds setting for launcher: `disableAutoSafeMode`
Updates UI to permit setting of `disableAutoSafeMode`
Updates unit test to ensure value is persisted

image

## Related Issue(s)

owner: #4920 
parent: https://github.com/FlowFuse/flowfuse/issues/4552

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [x] Suitable unit/system level tests have been added and they pass 
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `FlowFuse/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `FlowFuse/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

File: frontend/src/pages/instance/Settings/LauncherSettings.vue [hunk 1]
```diff
@@ -8,6 +8,13 @@
                 Flows that perform CPU intensive work may need to increase this from the default of 7500ms.
             </template>
         </FormRow>
+        <FormRow v-model="input.disableAutoSafeMode" type="checkbox">
+            Disable Auto Safe Mode
+            <template #description>
+                Prevent Node-RED from automatically entering safe mode when a crash loop is detected.
+                WARNING: Disabling Auto Safe Mode is not recommended. A problem that causes Node-RED to crash multiple successive times may result in a contineous bootloop that will need to be manually resolved.
+            </template>
+        </FormRow>
```
Comment (hardillb) [inline frontend/src/pages/instance/Settings/LauncherSettings.vue:17]: This is going to need a launcher version check so it's not shown to pre 2.12 launchers and a message about upgrading is shown.

See here for an example 
- https://github.com/FlowFuse/flowfuse/blob/main/frontend/src/pages/admin/Template/sections/NPMRegistry.vue
- https://github.com/FlowFuse/flowfuse/pull/4900/files#diff-b9045860ec8e01709bcc1244f38e95637e7dfdb7df7dd4e114954506137bcb16

Comment (Steve-Mcl) [reply frontend/src/pages/instance/Settings/LauncherSettings.vue:17]: > This is going to need a launcher version check so it's not shown to pre 2.12 launchers and a message about upgrading is shown.

ah, indeed. good call Ben.

Will push update soon.

**Comment by hardillb** [comment]
Also when I toggle the checkbox the "save settings" button is not being enabled (or changing the health check timeout)

**Comment by Steve-Mcl** [comment]
> Also when I toggle the checkbox the "save settings" button is not being enabled (or changing the health check timeout)
> 
> The `data` section doesn't appear to have been updated to include the new value so it's not getting tracked.

That was working for me, but good spot on missing data entries (now added)

File: frontend/src/pages/instance/Settings/LauncherSettings.vue [hunk 2]
```diff
@@ -8,6 +8,13 @@
                 Flows that perform CPU intensive work may need to increase this from the default of 7500ms.
             </template>
         </FormRow>
+        <FormRow v-if="launcherSupportsAutoSafeMode" v-model="input.disableAutoSafeMode" type="checkbox">
+            Disable Auto Safe Mode
+            <template #description>
+                Prevent Node-RED from automatically entering safe mode when a crash loop is detected.
+                WARNING: Disabling Auto Safe Mode is not recommended. A problem that causes Node-RED to crash multiple successive times may result in a contineous bootloop that will need to be manually resolved.
+            </template>
+        </FormRow>
```
Comment (hardillb) [inline frontend/src/pages/instance/Settings/LauncherSettings.vue:17]: This needs a `v-else` block afterwards to show the link to the upgrade instructions

**Comment by hardillb** [comment]
Toggling the checkbox is still now marking the save settings button as active

**Comment by hardillb** [comment]
@Steve-Mcl take hte 5000 out of the box at the top, the default is empty

**Comment by Steve-Mcl** [comment]
> @Steve-Mcl take hte 5000 out of the box at the top, the default is empty

Ah, ok. This was existing behaviour but now there is more than one setting on the form, it matters!

I'll modify the form to show the default health value so the validation works.

File: frontend/src/pages/instance/Settings/LauncherSettings.vue (hunk 2 above)
Comment (Steve-Mcl) [reply frontend/src/pages/instance/Settings/LauncherSettings.vue:17]: Done in incoming update:

image

**Comment by Steve-Mcl** [comment]
ok, fingers crossed, 3rd time is sure to be the charm.

I've added the v-else message that some settings are not available until stack is upgraded.

I set the local health value input to the launchers default (even if not set in the DB settings) - this way, there are no validation errors and the Auto Safe Mode toggle can be operated/changed/saved.

Lastly I made sure settings are only passed/updated if changed.

**Review by hardillb — APPROVED** [review]
But not merging as the updated stacks won't be available until release.

We should merge this on release day along with https://github.com/FlowFuse/flowfuse/pull/4900

**Pull request merged**